relay/README.md

```diff
@@ -24,18 +24,25 @@ libp2p already has modules for NAT ([go-libp2p-nat](https://github.com/libp2p/go
 but these don't always do the job, just because NAT traversal is complicated.
 That's why it's useful to have a simple relay protocol.
 
-One node asks a relay node to connect to another node on its behalf.
-The relay node shortcircuits its streams to the two nodes,
+Unlike a transparent **tunnel**, where a libp2p peer would just proxy a
+communication stream to a destination (the destination being unaware of the
+original source), a circuit-relay makes the destination aware of the original
+source and the circuit followed to establish communication between the two.
+This provides the destination side with full knowledge of the circuit which,
+if needed, could be rebuilt in the opposite direction using
```

Comment:
I think the key point that confused me and something worth highlighting here is that, the relay connection can be rebuilt using the same `relay` node or by asking the swarm to route it over some other relay peer. So I would perhaps word it as `if needed, could be rebuilt either through the same relay node, or if unavailable by asking the swarm to route the connection over any other known relay`.